### PR TITLE
WIP feat(breadcrumbs): Add Breadcrumbs component to Canvas Labs

### PR DESCRIPTION
**Pull request by emilyjablonski**
## Summary

This PR introduces a Breadcrumb component in Labs for use navigating item structures. It uses a Canvas element to size the list of breadcrumbs based on the container's width, and dynamically hides any that do not fit appropriately under an expander.

This PR is currently a WIP and will continue to be iterated upon.

## Checklist

- [x] branch has been rebased on the latest master commit
- [x] tests are changed or added
- [x] `yarn test` passes
- [ ] all (dev)dependencies that the module needs is added to its `package.json`
- [x] code has been documented and, if applicable, usage described in README.md
- [ ] module has been added to `canvas-kit-react` and/or `canvas-kit-css` universal modules, if
      applicable
- [ ] design approved final implementation
- [ ] a11y approved final implementation
- [ ] code adheres to the [API & Pattern guidelines](../API_PATTERN_GUIDELINES.md)

## Additional References
Screen Shot 2020-05-14 at 5 59 15 PM

### CODE REVIEW

File: modules/_labs/breadcrumbs/react/spec/Breadcrumbs.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    const alert = jest.spyOn(window, `alert`).mockImplementation(() => {}); // eslint-disable-line no-empty-function
+    const {findByTestId, findByText} = render(
+      <Breadcrumbs breadcrumbs={exampleBreadcrumbs} containerWidth={300} />
+    );
+
+    // Act:
+    const ellipsis = await waitForElement(() => findByTestId(`more-breadcrumbs`));
+    act(() => {
+      fireEvent.click(ellipsis);
+    });
```
Comment (NicholasBoll) [inline modules/_labs/breadcrumbs/react/spec/Breadcrumbs.spec.tsx:117]: You shouldn't need to use `act` in newer versions of React Testing Library

File: modules/_labs/breadcrumbs/react/spec/Breadcrumbs.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+  test(`dropdown calls action when clicked`, async () => {
+    // Arrange:
+    const alert = jest.spyOn(window, `alert`).mockImplementation(() => {}); // eslint-disable-line no-empty-function
+    const {findByTestId, findByText} = render(
+      <Breadcrumbs breadcrumbs={exampleBreadcrumbs} containerWidth={300} />
+    );
+
+    // Act:
+    const ellipsis = await waitForElement(() => findByTestId(`more-breadcrumbs`));
```
Comment (NicholasBoll) [inline modules/_labs/breadcrumbs/react/spec/Breadcrumbs.spec.tsx:114]: `findBy` already uses `waitForElement` and isn't needed here

File: modules/_labs/breadcrumbs/react/lib/DropdownPopover.tsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        props.setOpenState(false);
+        props.setActiveOption(dummyOption);
+        onOptionChange(option);
+        break;
+
+      case `ArrowUp`:
+      case `ArrowLeft`:
+      case `Up`:
+        handleUpdateActiveOption(optionList.indexOf(option), true);
+        break;
```
Comment (NicholasBoll) [inline modules/_labs/breadcrumbs/react/lib/DropdownPopover.tsx:160]: Should `Left` be added as well?

File: modules/_labs/breadcrumbs/react/lib/DropdownPopover.tsx [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+      case `ArrowDown`:
+      case `Down`:
+      case `ArrowRight`:
+        handleUpdateActiveOption(optionList.indexOf(option), false);
+        break;
```
Comment (NicholasBoll) [inline modules/_labs/breadcrumbs/react/lib/DropdownPopover.tsx:166]: Should `Right` be added as well?

File: modules/_labs/breadcrumbs/react/lib/Breadcrumbs.tsx [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+  const activeOptionEl: React.RefObject<HTMLDivElement> = React.useRef(null);
+  const dividerWidth = 42;
+  const breadcrumbsWidths: number[] = [];
+  let context: CanvasRenderingContext2D | null | undefined;
+
+  React.useLayoutEffect(() => {
+    if (activeOptionEl.current) {
+      activeOptionEl.current.focus();
+    }
+  });
```
Comment (alexandrzavalii) [inline modules/_labs/breadcrumbs/react/lib/Breadcrumbs.tsx:76]: no dependencies?

**Comment by lychyi** [comment]
@alanbsmith this can probably be closed now right?

**Comment by alanbsmith** [comment]
Yep! Breadcrumbs is now in CK-L! Thanks @emilyjablonski! 🎉

**Pull request closed without merging**